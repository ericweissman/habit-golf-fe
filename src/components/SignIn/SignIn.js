import React, { Component } from 'react';

export class SignIn extends Component {
  state = {
    email: '',
    password: '',
  }

  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value })
  }

  render() {
    const { email, password } = this.state
    const { newUser } = this.props
    return (
      <form className="sign-in-form">
        <div>
          <label>
            email
            <input type="text" name="email" value={email} onChange={this.handleChange} />
          </label>
          <label>
            password
            <input type="text" name="password" value={password} onChange={this.handleChange} />
          </label>
        </div>
        <button className="sign-in-btn">sign in</button>
        <button onClick={newUser} className='create-account-btn'>create account</button>
      </form>
    );
  }
}

export default SignIn;
