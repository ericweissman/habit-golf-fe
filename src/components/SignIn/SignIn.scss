.sign-in-form {
  background-color: $masters-green;
  width: 285px;
  display: flex;
  margin-top: 20px;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border-radius: 10px;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  div {
    display: flex;
    flex-direction: column;
  }
  label {
    font-weight: 200;
    color: $masters-yellow;
    font-size: 20px;
    padding: 10px 15px;
    margin-top: 20px;
  }
  input {
    outline: none;
    border-radius: 5px;
    height: 43px;
    font-size: 15px;
    font-weight: 200;
    padding-left: 10px;
    margin-left: 5px;
    width: 245px;
    margin-top: 10px;
  }
}

.sign-in-btn {
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  background-color: $masters-yellow;
  height: 50px;
  width: 125px;
  outline: none;
  border: none;
  border-radius: 50px;
  font-size: 20px;
  font-weight: 200;
  padding: 0 10px;
  margin-top: 20px;
  &:hover {
    font-weight: 700;
    color: $masters-green;
  }
}

.create-account-btn {
  @extend .sign-in-btn;
  width: 170px;
  margin-bottom: 20px;
}