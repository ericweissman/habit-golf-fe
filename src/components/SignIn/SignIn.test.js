import React from 'react';
import SignIn from './SignIn';
import { shallow } from 'enzyme'

describe('SignIn', () => {
  let wrapper
  it('should correctly match the snapshot', () => {
    wrapper = shallow(<SignIn />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should have the correct default state', () => {
    const expectedState = {
      email: '',
      password: '',
    }
    expect(wrapper.state()).toEqual(expectedState)
  })

  it('should correctly update state based on input', () => {
    const mockEvent1 = { target: { name: 'email', value: 'eric@geocities.com' } }
    const mockEvent2 = { target: { name: 'password', value: 'abc' } }
    const initialState = {
      email: '',
      password: '',
    }
    const expectedState = {
      email: 'eric@geocities.com',
      password: 'abc',
    }

    wrapper.setState(initialState)
    wrapper.instance().handleChange(mockEvent1);
    wrapper.instance().handleChange(mockEvent2);

    expect(wrapper.state()).toEqual(expectedState)
  })
})
