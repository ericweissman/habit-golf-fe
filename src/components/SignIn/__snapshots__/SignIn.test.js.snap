// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should correctly match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <label>
            email
            <input
              name="email"
              onChange={[Function]}
              type="text"
              value=""
            />
          </label>
          <label>
            password
            <input
              name="password"
              onChange={[Function]}
              type="text"
              value=""
            />
          </label>
        </div>,
        <button
          className="sign-in-btn"
        >
          sign in
        </button>,
        <button
          className="create-account-btn"
        >
          create account
        </button>,
      ],
      "className": "sign-in-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              email
              <input
                name="email"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>,
            <label>
              password
              <input
                name="password"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "email",
                <input
                  name="email"
                  onChange={[Function]}
                  type="text"
                  value=""
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "email",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "email",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "password",
                <input
                  name="password"
                  onChange={[Function]}
                  type="text"
                  value=""
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "password",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "password",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "sign in",
          "className": "sign-in-btn",
        },
        "ref": null,
        "rendered": "sign in",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "create account",
          "className": "create-account-btn",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "create account",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <label>
              email
              <input
                name="email"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>
            <label>
              password
              <input
                name="password"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>
          </div>,
          <button
            className="sign-in-btn"
          >
            sign in
          </button>,
          <button
            className="create-account-btn"
          >
            create account
          </button>,
        ],
        "className": "sign-in-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                email
                <input
                  name="email"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </label>,
              <label>
                password
                <input
                  name="password"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </label>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "email",
                  <input
                    name="email"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "email",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "email",
                    "onChange": [Function],
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "password",
                  <input
                    name="password"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "password",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "password",
                    "onChange": [Function],
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "sign in",
            "className": "sign-in-btn",
          },
          "ref": null,
          "rendered": "sign in",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "create account",
            "className": "create-account-btn",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "create account",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
